### ISSUE TITLE
508-defect-2 [FOCUS MANAGEMENT, SCREEN READER]: After editing/adding an address, focus MUST be returned to where user previously was

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#9944]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

When a user edits or adds an address for their shipping address, when they return to the Shipping address screen, the focus **must** return to where they triggered the action to edit or add an address.

At present, when the user is returned to the Shipping address screen, focus is on the `.saved-success-container`, and then flow proceeds to the link to "Finish this order later", then "Need help?". A non-sighted user would be confused, and miss the selection of shipping address, email address entry, and options to go back or continue with the form.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to select to edit/add an address, and when completed, return to where I initiated modifying that address so that I may continue to complete the form.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: all
* Server destination: staging

## Steps to Recreate

1. Enter https://staging.va.gov/hearing-aid-batteries-and-accessories/veteran-information/addresses in browser
2. Start screenreading device listed in Environment
3. Proceed through the form
4. Select to Edit permanent address
5. Edit the permanent address, and Save permanent address
6. Verify that focus is moved to the "Application has been saved.…"  notice beneath the form buttons
7. Verify that the rest of the form, to choose which address to ship to, add an email address, or progress back/continue with the form are not in the next steps available

## Possible Fixes (optional)

Return focus to the button that initiated editing/adding an address.

## WCAG or Vendor Guidance (optional)

* [Focus Order: Understanding SC 2.4.3](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html)

**Comment by mr0sari0** [comment]
Hi @jenstrickland, could you please review this again?

**Comment by jenstrickland** [comment]
@mr0sari0 — I just tested this in staging, but focus does not seem to be managed. When I select Edit this address in the Permanent address section, focus moved to the label for the checkbox for the military base addresses, and when I saved the address change, focus moved to the page breadcrumbs. 

Are the changes available in staging, yet?

**Comment by mr0sari0** [comment]
Hi @jenstrickland, the changes should be in staging now.

**Comment by jenstrickland** [comment]
Validated. 🎉
